src/sentry/integrations/discord/actions/notification.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Generator, Sequence
+
+from sentry.eventstore.models import GroupEvent
+from sentry.integrations.discord.actions.form import DiscordNotifyServiceForm
+from sentry.integrations.discord.client import DiscordClient
+from sentry.integrations.discord.message_builder.issues import DiscordIssuesMessageBuilder
+from sentry.rules.actions import IntegrationEventAction
+from sentry.rules.base import CallbackFuture, EventState
+from sentry.shared_integrations.exceptions.base import ApiError
+from sentry.types.rules import RuleFuture
+from sentry.utils import metrics
+
+
```

Comment:
The other big piece of code. This one is the logic that handles notification building and sending.